src/autorun.ts

```diff
@@ -0,0 +1,174 @@
+import { join, parse } from 'node:path';
+import fs from 'fs-extra';
+import { Type as T, Static } from '@sinclair/typebox';
+import { Value } from '@sinclair/typebox/value';
+import _ from 'lodash';
+import { readFileContent, readJsonObject, updateJsonObjectFile } from './fileUtils.js';
+
+//@ts-ignore
+import { Slangroom } from '@slangroom/core';
+//@ts-ignore
+import { wallet } from '@slangroom/wallet';
+//@ts-ignore
+import { http } from '@slangroom/http';
+
+import { config } from './cli.js';
+const L = config.logger;
+
+/* Main */
+
+const AUTORUN_CONTRACTS_PATH = join(config.zencodeDirectory, '.autorun');
+
+export async function autorunContracts() {
+	await autorunInstallContracts();
+	await autorunStartupContracts();
+}
+export async function autorunStartupContracts() {
+	const contracts = await loadContractsData(AUTORUN_CONTRACTS_PATH);
+	const startupContracts = contracts.filter(isStartupContract);
+	const contractsPromises = startupContracts.map(runContractData);
+	await Promise.allSettled(contractsPromises);
+}
+
+export async function autorunInstallContracts() {
+	const contracts = await loadContractsData(AUTORUN_CONTRACTS_PATH);
+	const installContracts = contracts.filter(isInstallContract);
+	const installContractsToRun = installContracts.filter(isInstallContractNotExecuted);
+	const contractsPromises = installContractsToRun.map(runInstallContractData);
+	await Promise.allSettled(contractsPromises);
+}
+
+/* Definitions */
+
+async function getContractPathsInDirectory(directoryPath: string): Promise<string[]> {
+	const directory = await fs.readdir(directoryPath);
+	return Array.from(directory.entries())
+		.map((entry) => entry[1])
+		.filter((name) => isZencodeFile(name))
+		.map((name) => join(directoryPath, name));
+}
+
+async function loadContractsData(directoryPath: string): Promise<ContractData[]> {
+	const contractPaths = await getContractPathsInDirectory(directoryPath);
+	const contractDataPromises = contractPaths.map(loadContractData);
+	const maybeContractData = await Promise.all(contractDataPromises);
+	return maybeContractData.filter((d) => Boolean(d)) as ContractData[];
+}
+
+async function runContractData({ contract, keys, conf, data, path }: ContractData) {
+	try {
+		const s = new Slangroom([http, wallet]);
```

Comment:
the lists of the plugins to load fro slangroom should be in common with the ones in src/index.ts please keep them in sync... should be nice to have maybe a Singleton? or a list of plugins as a constant to get the list as a single source of truth.

Reply:
I like the singleton approach
Fixed here https://github.com/ForkbombEu/ncr/pull/42/commits/7682f4c33d892a2325bdea8544395100ce47c948

Reply:
Lovely! A small detail (but can save junior devs time) is to not have the `Slangroom` from @slangroom/core and the Object of the singleton named in the same way, to prevent confusion! I bet that if you search by javistas naming singleton best practices, you'll find plenty of inspiration ;)